### PR TITLE
Setup CI with Travis and update setup script

### PR DESCRIPTION
**Pull request by bandang0**
For the CI, I suggest for now to test for PEP8 compliance and then to run `python3 setup.py test`.

Second thing: have the package requirements (only setuptools for now, is this really useful?) in a `requirements.txt` file.

### CODE REVIEW

**Comment by bandang0** [comment]
Ok @zuh0 this is now ready for you!

With this setup, once the user has pip installed, he or she will have the cli tool `aamg-gen` which triggers the `aamg` function formerly in `aamg/aamg.py`, with all relevant options, etc. Also, the user can `import aamg` and use the project simply as a library.

Of course, the `python setup.py test` still works as before.

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+language: python
```
Comment (gabriel-duque) [inline .travis.yml:1]: I'm not sure but there might be a way to test this on specific version expressions like `python>=3.6`.

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+# Default Python for Travis is *some* Python 3.
+# Install dependencies.
+install:
+    - pip install -r requirements.txt
+    - pip install pycodestyle
+# Run tests.
+script:
+    - pycodestyle .
```
Comment (gabriel-duque) [inline .travis.yml:9]: not sure what the default behavior is but it seems like `pycodestyle` have 2 nice options we might want to use:

```
  --show-source        show source code for each error
  --show-pep8          show text of PEP 8 for each error (implies --first)
```

File: aamg/generator.py [hunk 3]
```diff
@@ -2,7 +2,7 @@
 import random
 import typing
 
-import log
+from . import log
```
Comment (gabriel-duque) [inline aamg/generator.py:5]: Yes!

File: tests/lexer_test.py [hunk 4]
```diff
@@ -5,15 +5,11 @@
 import unittest
 import unittest.mock
 
-
-# We need this because our tests and source files are not in the same directory
-sys.path.insert(0, 'aamg')
-
-import generator
+from aamg import generator
```
Comment (gabriel-duque) [inline tests/lexer_test.py:8]: God I was tired!
That's the low-level C programmer "I know how my system works" me.

You are so right!

**Review by gabriel-duque — APPROVED** [review]
LGTM!

File: .travis.yml (hunk 1 above)
Comment (gabriel-duque) [reply .travis.yml:1]: to make sure our python version will handle f-strings

**Pull request merged**